runtime/lua/vim/lsp/util.lua

```diff
@@ -1772,7 +1788,13 @@ function M._get_offset_encoding(bufnr)
   local offset_encoding
 
   for _, client in pairs(vim.lsp.buf_get_clients(bufnr)) do
-    local this_offset_encoding = client.offset_encoding or "utf-16"
+    if client.offset_encoding == nil then
+      vim.notify_once(
+        string.format("Client (id: %s) offset_encoding is nil. Do not unset offset_encoding.", client.id),
+        vim.log.levels.WARN
+      )
+    end
```

Comment:
I think here it would be okay to turn this into an error.

Reply:
Done